checks.d/agent_metrics.py

```diff
@@ -0,0 +1,138 @@
+import psutil
+import os
+import threading
+from checks import AgentCheck
+from config import _is_affirmative
+
+MAX_THREADS_COUNT = 50
+MAX_COLLECTION_TIME = 30
+MAX_EMIT_TIME = 5
+MAX_CPU_PCT = 10
+
+
+class AgentMetrics(AgentCheck):
+    """
+    New-style version of `CollectorMetrics`
+    Gets information about agent performance on every collector loop
+    """
+
+    def __init__(self, *args, **kwargs):
+        AgentCheck.__init__(self, *args, **kwargs)
+        self._collector_payload = {}
+        self._metric_context = {}
+
+    def _psutil_config_to_stats(self, instance):
+        """
+        Reads `init_config` for `psutil` methods to call on the current process
+        Calls those methods and stores the raw output
+
+        :returns a dictionary of statistic_name: value
+        """
+        process_metrics = instance.get('process_metrics', self.init_config.get('process_metrics', None))
+        if not process_metrics:
+            self.log.error('No metrics configured for AgentMetrics check!')
+            return {}
+
+        current_process = psutil.Process(os.getpid())
+        filtered_methods = [k for k,v in process_metrics.items() if _is_affirmative(v) and\
+                                hasattr(current_process, k)]
+        stats = {}
+
+        if filtered_methods:
+            for method in filtered_methods:
+                # Go from `get_memory_info` -> `memory_info`
+                method_key = method[4:] if method.startswith('get_') else method
+                try:
+                    raw_stats = getattr(current_process, method)()
+                    try:
+                        stats[method_key] = raw_stats._asdict()
+                    except AttributeError:
+                        if isinstance(raw_stats, int):
+                            stats[method_key] = raw_stats
+                        else:
+                            self.log.warn("Could not serialize output of {} to dict".format(method))
```

Comment:
The `{}` syntax is not python 2.6 compatible... It should be named placeholders `{0}`.